frontend/src/screens/HomeScreen.js

```diff
@@ -1,15 +1,40 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
 import Product from '../components/Product';
-import data from '../data';
+import LoadingBox from '../components/LoadingBox';
+import MessageBox from '../components/MessageBox';
 
 export default function HomeScreen() {
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  useEffect(() => {
+    const fecthData = async () => {
+      try {
+        setLoading(true);
+        const { data } = await axios.get('/api/products');
+        setLoading(false);
+        setProducts(data);
+      } catch (err) {
+        setError(err.message);
+        setLoading(false);
+      }
+    };
+    fecthData();
+  }, []);
   return (
     <div>
-      <div className="row center">
-        {data.products.map((product) => (
-          <Product key={product._id} product={product}></Product>
-        ))}
-      </div>
+      {loading ? (
+        <LoadingBox></LoadingBox>
+      ) : error ? (
+        <MessageBox variant="danger">{error}</MessageBox>
+      ) : (
+        <div className="row center">
+          {products.map((product) => (
```

Comment:
×
TypeError: products.map is not a function

Reply:
use empty array in useState([])

Reply:
./src/screens/ProductScreen.js
Module not found: Can't resolve '../data' in 'E:\lapiro\SkillShare Training\amazona\frontend\src\screens'

Reply:
this is the error i get above pls i need assistance

Reply:
You probably moved the data.js file from the frontend/src folder to the backend folder instead of copying it. The ProductScreen.js file searches for data.js and doesn't find it, so that's why you get the error. To resolve this, copy the data.js file again to the frontend/src folder. While the two lessons the ProductScreen.js file will be updated then you can delete the data.js from the frontend.

Reply:
@mbilae 
You probably moved the data.js file from the frontend/src folder to the backend folder instead of copying it. The ProductScreen.js file searches for data.js and doesn't find it, so that's why you get the error. To resolve this, copy the data.js file again to the frontend/src folder. While the two lessons the ProductScreen.js file will be updated then you can delete the data.js from the frontend.